### PR TITLE
feat(core): Add configurable execution history limit

### PR DESCRIPTION
**Pull request by ahsanv**

### CODE REVIEW

**Review by netroy — DISMISSED** [review]

File: packages/cli/src/WorkflowExecuteAdditionalData.ts [hunk 1]
```diff
@@ -202,14 +203,29 @@ async function pruneExecutionData(this: WorkflowHooks): Promise<void> {
...
+		const maxCount = config.getEnv('executions.pruneDataCountMax');
 		const date = new Date(); // today
 		date.setHours(date.getHours() - maxAge);
 
 		// date reformatting needed - see https://github.com/typeorm/typeorm/issues/2286
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 		const utcDate = DateUtils.mixedDateToUtcDatetimeString(date);
 
-		const toPrune = { stoppedAt: LessThanOrEqual(utcDate) };
+		const toPrune: FindOptionsWhere<IExecutionFlattedDb> = { stoppedAt: LessThanOrEqual(utcDate) };
```
Comment (netroy) [inline packages/cli/src/WorkflowExecuteAdditionalData.ts:214]: I changed the typings on this line. If it looks good to you, please feel free to merge.

File: packages/cli/src/config/schema.ts [hunk 2]
```diff
@@ -325,6 +325,16 @@ export const schema = {
 			default: 3600,
 			env: 'EXECUTIONS_DATA_PRUNE_TIMEOUT',
 		},
+
+		// Additional pruning option to delete executions if total count exceeds the configured max.
+		// Deletes the oldest entries first
+		// Default is 0 = No limit
+		pruneDataCountMax: {
```
Comment (mieky) [inline packages/cli/src/config/schema.ts:332]: Nit: there's already `pruneDataMaxAge`, so for consistency between these two this ought to be `pruneDataMaxCount`

Comment (netroy) [reply packages/cli/src/config/schema.ts]: updated

**Review by mieky — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.217.1`

**Pull request merged**